plutus-core/plutus-core/src/PlutusCore/Constant/Typed.hs

```diff
@@ -459,24 +459,27 @@ We use @default@ for providing instances for built-in types instead of @Deriving
 latter breaks on @m a@
 -}
 
+-- TODO: in @makeKnown@ @cause@ and @term@ can be different, while in @readKnown@ they must be the
+-- same. We need to check if making them different in @readKnown@ is feasiable and can simplify
+-- things such as error handling in the CEK machine.
```

Comment:
Gonna try this next.

Reply:
Seems to have [worked](https://github.com/effectfully/plutus-prototype/commit/baaa34f40882a4aad44d0897a9ea5943631ad367) really well. If we're going to merge what's in this PR, then we should merge what's in that commit too. @michaelpj

Reply:
I agree, it looks great, add it to this PR.